### DISCUSSION TITLE
Styled Component Plugin & core mismatch?

### DISCUSSION THREAD

**Discussion by eMerzh** [swc-project/swc#9503] [Q&A] [answered]
Hello, 

I currently have swc + styled component plugin, and it works well :) 🙌 
I currently have : 
```
  "@swc/core": "1.7.18",
    "@swc/jest": "0.2.36",
    "@swc/plugin-styled-components": "2.0.9",
```

Now, i want to update to latest plugin version, either v2.0.12 or v3 .... but neither of those build...
they crash with message like

>Please ensure the version of `swc_core` used by the plugin is compatible with the host runtime

If i'm correct [v2.0.12](https://github.com/swc-project/plugins/blob/main/packages/styled-components/CHANGELOG.md#2012) state : "Update swc_core to v0.101.x"

so i need my core to be that version right? 

if i see correctly my 1.7.18 is https://github.com/swc-project/swc/commit/54f99a293d2f248be60eaedf0b29e384019530fc v0.101.6 so it should be ok no ? 

what am i missing?

Thanks for the help

**Comment by kdy1** [comment, marked as answer]
You may need to look at the lock file. Sometimes version in package.json does not match the actual installed version

**Reply by eMerzh** [reply]
hum... it seems there was still references of @swc/core@^1.5.7 in the lock 🤨  ... removed the lock file, and it tried again, it works now...
i have to understand why 🤔 

Thanks for the help 🙌
